src/Collection.php

```diff
@@ -108,10 +107,11 @@ public function __construct($collection, $entityRoute = null, $entityRouteParams
             throw new InvalidCollectionException(sprintf(
                 '%s expects an array or Traversable; received "%s"',
                 __METHOD__,
-                is_object($collection) ? get_class($collection) : gettype($collection)
+                is_object($collection) ? $collection::class : gettype($collection)
             ));
         }
 
+        /** @psalm-var Paginator|Traversable|array<array-key, mixed> $collection */
```

Comment:
Hi @gsteel, for some reason Psalm has lots of problems identifying some variable types even if they are already defined. This is why I had to add some of them as `@psalm-var,` because adding the types as `@var` was a useless duplication reported by PHPCS.

Reply:
The problem with forcing the type with `@psalm-var` is that it kills off Psalm's type inference - agreed that sometimes there's no other option, but in this case, adjusting both the method parameter and the property declaration to the correct type should negate the need for the forced type.

You'll probably find that the type safety condition above will produce a new psalm issue for `RedundantConditionGivenDocBlockType` once the type annotations have been corrected - typically, this issue should go in the baseline because the check is part of the class behaviour and removing it would be a BC break.